pkgs/servers/consul/default.nix

```diff
@@ -1,30 +1,20 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "consul";
-  version = "1.6.2";
+  version = "1.7.2";
   rev = "v${version}";
 
-  goPackagePath = "github.com/hashicorp/consul";
-
-  # Note: Currently only release tags are supported, because they have the Consul UI
-  # vendored. See
-  #   https://github.com/NixOS/nixpkgs/pull/48714#issuecomment-433454834
-  # If you want to use a non-release commit as `src`, you probably want to improve
-  # this derivation so that it can build the UI's JavaScript from source.
-  # See https://github.com/NixOS/nixpkgs/pull/49082 for something like that.
-  # Or, if you want to patch something that doesn't touch the UI, you may want
-  # to apply your changes as patches on top of a release commit.
```

Comment:
Could you comment (here, or even better in the commit message) whether this restriction is now lifted?

Reply:
No idea, but in general this is universally true for all software. If you try and build at a broken commit it won't work - if consul fails to check in js assets and it fails, then the commit was bad. I don't see why we need to call this out for advanced users who are modifying nix expressions to change which revision something builds at.

Also the linked PR doesn't have anything to do w/ consul - that's vault (which is by the same people), but is definitely a different piece of software.

Reply:
I don't think this is a big deal though, added the comments back.

Reply:
> No idea, but in general this is universally true for all software.

That is not my experience -- for most projects the release tarballs are just snapshots of the actual repository, and it is very common for nix users to modify `src` in `overrideAttrs` with a newer version from git; consul is behaving particularly badly here, which is why I found this comment helpful.

> Also the linked PR doesn't have anything to do w/ consul - that's vault

The PR diff is about vault, but most nixpkgs maintainers discussion about the consul UI has been in that thread (search for `consul` on the page), that's why got linked.

> I don't think this is a big deal though, added the comments back.

Thanks!